File: eco-drive-symfony/src/Document/Vehicle.php
<?php

namespace App\Document;

use Doctrine\ODM\MongoDB\Mapping\Annotations\Document;
use Doctrine\ODM\MongoDB\Mapping\Annotations\Id;
use Doctrine\ODM\MongoDB\Mapping\Annotations\Field;
use Doctrine\ODM\MongoDB\Mapping\Annotations\EmbedOne;
use Doctrine\ODM\MongoDB\Mapping\Annotations\EmbedMany;

#[Document(collection: "vehicles")]
class Vehicle
{
    #[Id]
    private ?string $id = null;

    #[Field(type: "string")]
    private ?string $brand = null;

    #[Field(type: "string")]
    private ?string $status = null;

    #[EmbedOne(targetDocument: Model::class)]
    private ?Model $model = null;

    #[Field(type: "string")]
    private ?string $registration = null;

    #[EmbedMany(targetDocument: Incident::class)]
    private array $incidentHistories = [];

    public function getId(): ?string
    {
        return $this->id;
    }

    public function setId(?string $id): void
    {
        $this->id = $id;
    }

    public function getBrand(): ?string
    {
        return $this->brand;
    }

    public function setBrand(?string $brand): void
    {
        $this->brand = $brand;
    }

    public function getStatus(): ?string
    {
        return $this->status;
    }

    public function setStatus(?string $status): void
    {
        $this->status = $status;
    }

    public function getModel(): ?Model
    {
        return $this->model;
    }

    public function setModel(?Model $model): void
    {
        $this->model = $model;
    }

    public function getRegistration(): ?string
    {
        return $this->registration;
    }

    public function setRegistration(?string $registration): void
    {
        $this->registration = $registration;
    }

    public function getIncidentHistories(): array
    {
        return $this->incidentHistories;
    }

    public function setIncidentHistories(array $incidentHistories): void
    {
        $this->incidentHistories = $incidentHistories;
    }
}

File: spring-boot-starter-data-rest/src/main/java/fr/dawan/formation/models/Vehicle.java
package fr.dawan.formation.models;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import java.util.ArrayList;
import java.util.List;

@Document(collection = "vehicles")
public class Vehicle {
    @Id
    private String id;
    private String brand;
    private String status;
    private Model model;
    private String registration;
    @Field("incident_histories")
    private List<Incident> incidentHistories = new ArrayList<>();

    public Vehicle() {
    }

    public Vehicle(String id, String brand, String status, Model model, String registration, List<Incident> incidentHistories) {
        this.id = id;
        this.brand = brand;
        this.status = status;
        this.model = model;
        this.registration = registration;
        this.incidentHistories = incidentHistories;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Model getModel() {
        return model;
    }

    public void setModel(Model model) {
        this.model = model;
    }

    public String getRegistration() {
        return registration;
    }

    public void setRegistration(String registration) {
        this.registration = registration;
    }

    public List<Incident> getIncidentHistories() {
        return incidentHistories;
    }

    public void setIncidentHistories(List<Incident> incidentHistories) {
        this.incidentHistories = incidentHistories;
    }

    @Override
    public String toString() {
        return "Vehicle{" +
            "id=" + id +
            ", brand='" + brand + '\'' +
            ", status='" + status + '\'' +
            ", model=" + model +
            ", registration='" + registration + '\'' +
            ", incidentHistories=" + incidentHistories +
            '}';
    }
}
